core/src/test/scala/org/apache/spark/rdd/PipedRDDSuite.scala

```diff
@@ -51,6 +51,22 @@ class PipedRDDSuite extends SparkFunSuite with SharedSparkContext {
     }
   }
 
+  test("basic pipe with tokenization") {
+    if (testCommandAvailable("wc")) {
+      val nums = sc.makeRDD(Array(1, 2, 3, 4), 2)
+
+      // verify that both RDD.pipe(command: String) and RDD.pipe(command: String, env) work good
+      for (piped <- Seq(nums.pipe("wc -l"), nums.pipe("wc -l", Map[String, String]()))) {
```

Comment:
You know, we actually had no tests testing `pipe(String)`! This is one.
Feel free to add even the most basic tests exercising this method as it would have caught the underlying error that caused the infinite recursion.